channels/generic/websockets.py

```diff
@@ -162,11 +162,58 @@ def send(self, content, close=False):
         """
         Encode the given content as JSON and send it to the client.
         """
-        super(JsonWebsocketConsumer, self).send(text=json.dumps(content), close=close)
+        super(JsonWebsocketConsumer, self).send(text=self.encode_json(content), close=close)
+
+    @classmethod
+    def decode_json(cls, text):
+        return json.loads(text)
+
+    @classmethod
+    def encode_json(cls, content):
+        return json.dumps(content)
 
     @classmethod
     def group_send(cls, name, content, close=False):
-        WebsocketConsumer.group_send(name, json.dumps(content), close=close)
+        WebsocketConsumer.group_send(name, cls.encode_json(content), close=close)
+
+
+class WebsocketMultiplexer(object):
+    """
+    The opposite of the demultiplexer, to send a message though a multiplexed channel.
+
+    The multiplexer object is passed as a kwargs to the consumer when the message is dispatched.
+    This pattern allows the consumer class to be independent of the stream name.
+    """
+
+    stream = None
+    reply_channel = None
+
+    def __init__(self, stream, reply_channel):
+        self.stream = stream
+        self.reply_channel = reply_channel
+
+    def send(self, payload):
+        """Multiplex the payload using the stream name and send it."""
+        self.reply_channel.send(self.encode(self.stream, payload))
+
+    @classmethod
+    def encode_json(cls, content):
+        return json.dumps(content, cls=DjangoJSONEncoder)
+
+    @classmethod
+    def encode(cls, stream, payload):
+        """
+        Encodes stream + payload for outbound sending.
+        """
+        content = {"stream": stream, "payload": payload}
+        return {"text": cls.encode_json(content)}
+
+    @classmethod
+    def group_send(cls, name, stream, payload, close=False):
+        message = cls.encode(stream, payload)
```

Comment:
... except I changed `WebsocketMultiplexer.encode` to `cls.encode` here to keep things polymorphic.